connectors/open_policy_agent/go/api_default.go

```diff
@@ -13,18 +13,109 @@
 package openapiserver
 
 import (
+	"encoding/json"
+	"fmt"
 	"io/ioutil"
 	"log"
 	"net/http"
+	"os"
+	"strconv"
 
+	opabl "fybrik.io/fybrik/connectors/open_policy_agent/lib"
+	openapiclientmodels "fybrik.io/fybrik/pkg/taxonomy/model/base"
 	"github.com/gin-gonic/gin"
 )
 
+func getEnv(key string) string {
+	value, exists := os.LookupEnv(key)
+	if !exists {
+		log.Fatalf("Env Variable %v not defined", key)
+	}
+	log.Printf("Env. variable extracted: %s - %s\n", key, value)
+	return value
+}
+
+func constructPolicyManagerRequest(inputString string) (*openapiclientmodels.PolicyManagerRequest, error) {
+	log.Println("inconstructPolicymanagerRequest")
+	log.Println("inputString")
+	log.Println(inputString)
+	var input openapiclientmodels.PolicyManagerRequest
+	err := json.Unmarshal([]byte(inputString), &input)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("input: %v", input)
+
+	return &input, nil
+}
+
+func displayReq(input2 openapiclientmodels.PolicyManagerRequest) error {
+	bytes, err := input2.MarshalJSON()
+	if err != nil {
+		log.Println("err: ", err)
+		return nil
+	}
+	log.Println("marshalled request:", string(bytes))
+	return nil
+}
```

Comment:
I believe we implemented such debug functions several times already. I'm not even sure if it's needed and printing with %+v would be enough but in any case a function like that should move to `pkg/` and be reused across the project. Look if we have an implementation already in manage (cc @shlomitk1). 

I'm personally not a fan of these functions since they require marshalling just for debugging. I would prefer to use a gin middleware instead for logging both request and response and add a flag to either add the middleware or not.

Reply:
@roee88 is it ok to use this library [render](https://github.com/luci/go-render) as suggested [here](https://stackoverflow.com/questions/43259971/golang-how-to-print-struct-value-with-pointer) ? Currently I have used this library in the opa connector as shown [here](https://github.com/rohithdv/fybrik/blob/186fcd6c2bc818754c825f777e9fe0843c5eaf27/connectors/open_policy_agent/go/api_default.go#L57).

or do you suggest some other way for printing structs with pointers ? Please advise. Thanks.